helpers/legacyStore.js

```diff
@@ -6,13 +6,31 @@ var assign = require("can-util/js/assign/assign");
 
 module.exports = function (count, make, filter) {
 	/*jshint eqeqeq:false */
-	var getUniqueId = (function () {
+	var nextUniqueId = (function () {
 		var i = 0;
 		return function () {
 			return i++;
 		}
 	})();
 
+	// Check to see if an existing item has the ID we want to assign
+	var isIdUnique = function (items, id) {
+		for (var key in items) {
```

Comment:
I'll use `can-util`'s `each`, I do not think we can make the assumption it is list-like based on the current code and `each` will handle the object case.